tests/table_api_tests.py

```diff
@@ -0,0 +1,220 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+"""Unit tests for Superset"""
+import json
+
+from superset import db
+from superset.connectors.sqla.models import SqlaTable
+
+from .base_tests import SupersetTestCase
+
+
+class TableApiTests(SupersetTestCase):
+    resource_name = "table"
+
+    def __init__(self, *args, **kwargs):
+        super(TableApiTests, self).__init__(*args, **kwargs)
+
+    def _get_table(self, user="admin"):
+        """Helper function to send GET /table​/ request with given user and verify it succeeds."""
+        self.login(username=user)
+        uri = f"api/v1/table/"
+        rv = self.client.get(uri)
+        self.assertEqual(rv.status_code, 200)
+        data = json.loads(rv.data.decode("utf-8"))
+        self.assertEqual(data["count"], 6)
+
+    def _create_table(self, user="admin"):
+        """Helper function to send POST /table/ request with given user and verify it succeeds."""
+        # Create a table via POST /table/ API and verify it succeeds with 201 CREATED.
+        tbl_name = "bart_lines"
+        self.login(username=user)
+        table_data = {"database": 1, "table_name": tbl_name}
+        uri = f"api/v1/table/"
+        rv = self.client.post(uri, json=table_data)
+        self.assertEqual(rv.status_code, 201)
+        tbl_obj = self.get_table_by_name(tbl_name)
+        self.assertIsNotNone(tbl_obj)
```

Comment:
not needed, self.assertEqual(tbl_obj.table_name, tbl_name) will fail if `tbl_obj` is none